python/ray/data/dataset.py

```diff
@@ -4563,7 +4565,29 @@ def stats(self) -> str:
     def _get_stats_summary(self) -> DatasetStatsSummary:
         return self._plan.stats_summary()
 
-    @ConsumptionAPI(pattern="Time complexity:")
+    @ConsumptionAPI(pattern="")
+    @DeveloperAPI
+    def iter_internal_block_refs(self) -> Iterator[ObjectRef[Block]]:
+        """Get an iterator over references to the underlying blocks of this Dataset.
+
+        This function can be used for zero-copy access to the data. It does not
+        keep the data materialized in-memory.
```

Comment:
What does zero-copy access mean here? You might copy the data when you get the block reference, right?

Reply:
i had thought that when we get the `RefBundle` / `BlockRef`, it does not copy the data. that's the main advantage of passing the references instead of blocks themselves, right?

Reply:
Oh. Yeah, if you don't call `ray.get` there won't be any copies, although the way I read this makes it sound like I can access the actual `Block` without copies.

Reply:
good point, let me just remove the line. i think saying "It does not keep the data materialized in-memory." is the more important main point to get across.